### PR TITLE
feat: make rate_matching degradation factors configurable

### PR DESCRIPTION
**Pull request by liyuanzhe1991**
Convert module-level constants _RATE_MATCHING_PREFILL_DEGRADATION_FACTOR and _RATE_MATCHING_DECODE_DEGRADATION_FACTOR into configurable instance attributes on DisaggInferenceSession with a dedicated setter method. Propagate these parameters through TaskConfig.advanced_tuning_config and disagg_pareto() kwargs, eliminating the need for monkey-patching.
Add unit tests covering default values, setter behavior, and end-to-end parameter forwarding from TaskConfig to disagg_pareto.
Made-with: Cursor
#### Overview:
Previously, `_RATE_MATCHING_PREFILL_DEGRADATION_FACTOR` (0.9) and `_RATE_MATCHING_DECODE_DEGRADATION_FACTOR` (0.92) were module-level constants in `picking.py`, imported at module load time by `inference_session.py`. This made them impossible to override at runtime via monkey-patching—experiment scripts that attempted `picking._RATE_MATCHING_*_DEGRADATION_FACTOR = 1.0` had no effect on already-imported copies inside `DisaggInferenceSession`.
This PR converts these constants into configurable instance attributes with a proper setter API (`set_rate_matching_degradation_factors`), and threads the values through `TaskConfig.advanced_tuning_config` → `TaskRunner.run_disagg()` → `disagg_pareto()` → `DisaggInferenceSession`, making them fully configurable without monkey-patching.
#### Details:
- **`inference_session.py`**: Add `_rate_matching_prefill_degradation_factor` and `_rate_matching_decode_degradation_factor` as instance attributes on `DisaggInferenceSession.__init__` (defaulting to the module constants). Add `set_rate_matching_degradation_factors()` setter. Replace all internal references to the module constants with `self._rate_matching_*` attributes in `_get_disagg_summary_df` and `_find_best_result_under_constraints`.
- **`pareto_analysis.py`**: `disagg_pareto()` now accepts `rate_matching_prefill_degradation_factor` and `rate_matching_decode_degradation_factor` via `**kwargs`, and calls `disagg_sess.set_rate_matching_degradation_factors()` when either is provided.
- **`task.py`**: Add `rate_matching_prefill_degradation_factor: None` and `rate_matching_decode_degradation_factor: None` to the default `advanced_tuning_config`. `TaskRunner.run_disagg()` forwards these values to `disagg_pareto()`.
- **`test_inference_session.py`**: Add `TestRateMatchingDegradationFactors` class (5 tests): default values, setter with both/partial args, and end-to-end impact on `tokens/s/gpu` output.
- **`test_task.py`**: Add default-value assertions in `test_taskconfig_disagg_default`. Add `TestRateMatchingFactorsForwarding` class (3 tests): None forwarding, custom values, and partial override.
#### Where should the reviewer start?
- `src/aiconfigurator/sdk/inference_session.py` — the core change: new instance attributes and setter method (lines 175–202), and the two call sites that now use `self._rate_matching_*` (lines 220–221, 765–766).
- `src/aiconfigurator/sdk/pareto_analysis.py` — kwargs extraction and conditional setter call (lines 239–247).
- `src/aiconfigurator/sdk/task.py` — default config addition (lines 494–495) and forwarding in `run_disagg` (lines 1346–1351).
#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
- Relates to: disagg rate-matching configurability improvement

## Summary by CodeRabbit

* **New Features**
  * Added customizable rate-matching degradation factors for disaggregated inference optimization. Users can now configure prefill and decode degradation factors through task configuration or runtime API calls to fine-tune performance analysis.

### CODE REVIEW

File: src/aiconfigurator/sdk/pareto_analysis.py [hunk 1]
```diff
@@ -236,6 +236,16 @@ def get_working_list(working_list, max_constraint):
     disagg_sess = DisaggInferenceSession(prefill_database, prefill_backend, decode_database, decode_backend)
     disagg_sess.set_latency_correction_scales(prefill_latency_correction_scale, decode_latency_correction_scale)
 
+    rate_matching_prefill = kwargs.pop("rate_matching_prefill_degradation_factor", None)
```
Comment (tianhaox) [inline src/aiconfigurator/sdk/pareto_analysis.py:244]: should we make it to 1.0 by default?

Comment (liyuanzhe1991) [reply src/aiconfigurator/sdk/pareto_analysis.py:244]: I think it's better to keep it as what we set before, in case cause other alignment issuse

Comment (Arsene12358) [reply src/aiconfigurator/sdk/pareto_analysis.py:244]: As discussed, @liyuanzhe1991 let's add a comment telling people that **None** would actually mean we use our internally tunned default values

**Review by Arsene12358 — APPROVED** [review]
